AiiDA fails on concurrent sandbox creation

This was reported by @chrisjsewell, see here:
https://github.com/aiidateam/aiida-core/issues/4899#issuecomment-830133804

The error message is:
```
04/30/2021 02:22:43 PM <1613> aiida.engine.transports: [ERROR] Exception whilst using transport:
Traceback (most recent call last):
  File "/root/aiida-core/aiida/engine/transports.py", line 110, in request_transport
    yield transport_request.future
  File "/root/aiida-core/aiida/engine/processes/calcjobs/tasks.py", line 82, in do_upload
    with SandboxFolder() as folder:
  File "/root/aiida-core/aiida/common/folders.py", line 421, in __init__
    os.makedirs(sandbox)
  File "/usr/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/root/.aiida/repository/default/sandbox'

04/30/2021 02:22:43 PM <1613> aiida.orm.nodes.process.calculation.calcjob.CalcJobNode: [ERROR] iteration 1 of do_upload excepted, retrying after 20 seconds
Traceback (most recent call last):
  File "/root/aiida-core/aiida/engine/utils.py", line 188, in exponential_backoff_retry
    result = await coro()
  File "/root/aiida-core/aiida/engine/processes/calcjobs/tasks.py", line 82, in do_upload
    with SandboxFolder() as folder:
  File "/root/aiida-core/aiida/common/folders.py", line 421, in __init__
    os.makedirs(sandbox)
  File "/usr/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/root/.aiida/repository/default/sandbox'
```

This has been tested on AiiDA post-1.6.3, after the repository migration (even if a similar issue might have been there even before).

--- a/aiida/common/folders.py
+++ b/aiida/common/folders.py
@@ -342,8 +342,7 @@ class Folder:
         It is always safe to call it, it will do nothing if the folder
         already exists.
         """
-        if not self.exists():
-            os.makedirs(self.abspath, mode=self.mode_dir)
+        os.makedirs(self.abspath, mode=self.mode_dir, exist_ok=True)
 
     def replace_with_folder(self, srcdir, move=False, overwrite=False):
         """This routine copies or moves the source folder 'srcdir' to the local folder pointed to by this Folder.
@@ -370,8 +369,7 @@ class Folder:
 
         # Create parent dir, if needed, with the right mode
         pardir = os.path.dirname(self.abspath)
-        if not os.path.exists(pardir):
-            os.makedirs(pardir, mode=self.mode_dir)
+        os.makedirs(pardir, mode=self.mode_dir, exist_ok=True)
 
         if move:
             shutil.move(srcdir, self.abspath)
@@ -417,8 +415,7 @@ class SandboxFolder(Folder):
         # First check if the sandbox folder already exists
         if sandbox_in_repo:
             sandbox = os.path.join(get_profile().repository_path, 'sandbox')
-            if not os.path.exists(sandbox):
-                os.makedirs(sandbox)
+            os.makedirs(sandbox, exist_ok=True)
             abspath = tempfile.mkdtemp(dir=sandbox)
         else:
             abspath = tempfile.mkdtemp()

